src/black/__init__.py

```diff
@@ -626,8 +644,6 @@ def get_sources(
                     gitignore,
                 )
             )
-        elif s == "-":
-            sources.add(p)
         elif p.is_file():
```

Comment:
BTY `--exclude` isn't checked here as it is only checked against recursively found files and directories so the help for your new option is a bit misleading. Only `--force-exclude` is checked.

```console
(black) ichard26@acer-ubuntu:~/programming/oss/black$ echo "print('hello world')" | black -v --force-exclude="hello.*" --stdin-filename="hello.py" -
Using configuration from /home/ichard26/programming/oss/black/pyproject.toml.
hello.py ignored: matches the --force-exclude regular expression
No Python files are present to be formatted. Nothing to do 😴
(black) ichard26@acer-ubuntu:~/programming/oss/black$ echo "print('hello world')" | black -v --exclude="hello.*" --stdin-filename="hello.py" -
Using configuration from /home/ichard26/programming/oss/black/pyproject.toml.
print("hello world")
reformatted __BLACK_STDIN_FILENAME__/hello.py
All done! ✨ 🍰 ✨
1 file reformatted.
```
**NOTE**: the suggestion above must be applied for this demo to work

Reply:
Yeah, you are right! Since this is a solution that passing the path itself, it should not enforce exclude but only force-exclude.

Will make the changes together with the change on README and tests.